torch/_inductor/codegen/cpp.py

```diff
@@ -5346,34 +5372,48 @@ def max_parallel_depth(self):
         1) Levels without splitting and
         2) All reduction or non-reduction levels
         When the loop is split at the top level, the max depth is 1.
+        When the range of the first inner loop is much larger than the range of all outer loops,
+        change the starting depth of parallelization to the first inner loop.
```

Comment:
This also has the requirement on the reduction loop levels, i.e., all or none reduction from "start depth". Shall we document this here too?